v2/yarpcerror/errors.go

```diff
@@ -25,31 +25,52 @@ import (
 	"fmt"
 )
 
-// Newf returns a new Status.
+// Newf returns a new Status, with message formatting.
 //
 // The Code should never be CodeOK, if it is, this will return nil.
-func Newf(code Code, format string, args ...interface{}) *Status {
+func Newf(code Code, format string, args ...interface{}) error {
 	if code == CodeOK {
 		return nil
 	}
-	return &Status{
+	return New(code, sprintf(format, args...))
+}
+
+// New return a new Status.
+//
+// The Code should never be CodeOK; if it is, this will return nil.
+func New(code Code, message string, options ...StatusOption) error {
+	if code == CodeOK {
+		return nil
+	}
+	status := &status{
 		code:    code,
-		message: sprintf(format, args...),
+		message: message,
 	}
+	for _, opt := range options {
+		opt.apply(status)
+	}
+
+	// The name must only contain lowercase letters from a-z and dashes (-), and
+	// cannot start or end in a dash. If the name is something else, an error with
+	// code CodeInternal will be returned.
+	if err := validateName(status.name); err != nil {
+		return err
+	}
+	return status
 }
 
 // FromError returns the Status for the provided error. If the provided error
 // is not a Status, a new error with code CodeUnknown is returned.
 //
 // Returns nil if the provided error is nil.
-func FromError(err error) *Status {
+func FromError(err error) *status {
```

Comment:
We'll need to revisit this use case / maybe return an interface here? 🤔

Reply:
I don't quite follow. This is supposed to be a helper function to cast the error back to a status, and without this, there's no way to do so. Unless.. we expose the `Error` interface that Kris wants. But wasn't that the whole debate we had on Thursday?